fe/fe-core/src/main/java/com/starrocks/catalog/mv/MVTimelinessArbiter.java

```diff
@@ -162,10 +164,33 @@ protected Map<Table, Set<String>> collectBaseTableUpdatePartitionNames(Map<Table
         for (Map.Entry<Table, Column> e : refBaseTableAndColumns.entrySet()) {
             Table baseTable = e.getKey();
             MvBaseTableUpdateInfo mvBaseTableUpdateInfo = getMvBaseTableUpdateInfo(mv, baseTable,
-                    true, true);
+                    true, isQueryRewrite);
             mvUpdateInfo.getBaseTableUpdateInfos().put(baseTable, mvBaseTableUpdateInfo);
+            // If base table is a mv, its to-update partitions may not be created yet, skip it
             baseChangedPartitionNames.put(baseTable, mvBaseTableUpdateInfo.getToRefreshPartitionNames());
         }
         return baseChangedPartitionNames;
     }
-}
+
+    /**
+     * If base table is materialized view, add partition name to cell mapping into base table partition mapping;
+     * otherwise base table(mv) may lose partition names of the real base table changed partitions.
+     * @param baseTableUpdateInfoMap base table update info from MvTimelinessInfo
+     * @return the base table to its changed partition and cell map if it's mv, empty else
+     */
+    protected Map<Table, Map<String, PCell>> collectExtraBaseTableChangedPartitions(
+            Map<Table, MvBaseTableUpdateInfo> baseTableUpdateInfoMap) {
+        Map<Table, Map<String, PCell>> result = Maps.newHashMap();
```

Comment:
It's fine since `Table` 's hashcode/equals  methods use table id, or the full-qualified table name catalog.database.table as expected.

Table(OlapTable)
```
    @Override
    public int hashCode() {
        return Long.hashCode(id);
    }

    @Override
    public boolean equals(Object other) {
        if (!(other instanceof Table)) {
            return false;
        }
        Table otherTable = (Table) other;
        return id == otherTable.id;
    }
```

IcebergTable:
```
    @Override
    public int hashCode() {
        return com.google.common.base.Objects.hashCode(getCatalogName(), remoteDbName, getTableIdentifier());
    }

    @Override
    public boolean equals(Object other) {
        if (!(other instanceof IcebergTable)) {
            return false;
        }

        IcebergTable otherTable = (IcebergTable) other;
        String catalogName = getCatalogName();
        String tableIdentifier = getTableIdentifier();
        return Objects.equal(catalogName, otherTable.getCatalogName()) &&
                Objects.equal(remoteDbName, otherTable.remoteDbName) &&
                Objects.equal(tableIdentifier, otherTable.getTableIdentifier());
    }


```